torch/csrc/jit/codegen/cuda/lower_trivial_broadcast.cpp

```diff
@@ -0,0 +1,176 @@
+      auto p2c_map =
+          PairwiseRootDomainMap(producer, consumer)
+              .mapProducerToConsumer(
+                  producer->domain(), consumer->domain(), producer_broadcasts);
+      for (const auto& kv : p2c_map) {
+        auto p_id = kv.first;
+        auto c_id = kv.second;
+        const bool is_concretized = !c_id->isBroadcast();
+        auto it = broadcast_origin_map_.find(p_id);
+        TORCH_INTERNAL_ASSERT(
+            it != broadcast_origin_map_.end(),
+            "Broadcast origin info not found for producer broadcast domain: ",
+            p_id->toString(),
+            " of ",
+            producer->toString());
+        const auto& producer_origins = it->second;
+        if (is_concretized) {
+          // Keep track of all the origin domains as concretized
+          for (auto origin : producer_origins) {
+            // concretized_root_domains_.insert(origin);
+            markAsConcretized(origin);
+          }
+        } else {
+          // Not concretized yet. Propagate forward the origin info.
+          auto& consumer_origins = broadcast_origin_map_[c_id];
+          for (auto origin : producer_origins) {
+            consumer_origins.insert(origin);
+          }
+          consumer_origins.insert(c_id);
+        }
+      }
+    }
+  }
+}
+
+void ConcretizedBroadcastDomains::markAsConcretized(IterDomain* root_domain) {
+  concretized_domains_.emplace(root_domain);
+  std::deque<IterDomain*> child_domains({root_domain});
+  while (!child_domains.empty()) {
+    auto child = child_domains.front();
+    child_domains.pop_front();
+    const auto& child_uses = child->uses();
+    if (child_uses.size() == 1 && !child_uses[0]->outputs().empty() &&
+        std::all_of(
+            child_uses[0]->outputs().begin(),
+            child_uses[0]->outputs().end(),
+            [](const Val* out) { return out->isA<IterDomain>(); })) {
+      for (auto out : child_uses[0]->outputs()) {
+        child_domains.push_back(out->as<IterDomain>());
+      }
+    } else {
+      // leaf_domains.push_back(child);
+      concretized_domains_.emplace(child);
+    }
+  }
+}
+
+namespace {
+
+//! Traverse and collect all trivial broadcast expressions
+class TrivialBroadcastFinder : private IterVisitor {
+ public:
+  static std::unordered_set<BroadcastOp*> run(Fusion* fusion) {
+    TrivialBroadcastFinder finder(fusion);
+    return finder.trivial_bops_;
+  }
+
+ private:
+  TrivialBroadcastFinder(Fusion* fusion) {
+    traverse(fusion);
+  }
+
+  using IterVisitor::handle;
+
+  //! Check if a broadcast op is trivial.
+  void handle(BroadcastOp* bop) final {
+    const auto& pred_map = GpuLower::current()->threadPredMap();
+    auto out_tv = bop->out()->as<TensorView>();
+    // ConcretizedBroadcastDomains is used inside
+    // getParallelBroadcastDomains so non-concrete domains are not returned.
+    const auto parallel_types = pred_map.getParallelBroadcastDomains(out_tv);
```

Comment:
nit on the name: `parallel_broadcast_ids`

Reply:
Code was removed based on https://github.com/csarofeen/pytorch/pull/1412#discussion_r794234903